DEM.Net.Core/Services/Elevation/ElevationService.cs

```diff
@@ -557,8 +557,29 @@ public bool IsBoundingBoxCovered(BoundingBox bbox, IEnumerable<BoundingBox> bbox
             if (bboxTiles == null || !bboxTiles.Any())
                 return false;
 
-            // TODO: Union all the bboxTiles geometries (call it G) to see if G Union Bbox (or exclusive diff)
-            return true;
+            var factory = new GeometryFactory(new PrecisionModel(PrecisionModels.FloatingSingle));
+
+            ILinearRing bboxToLinearRing(BoundingBox boundingBox)
+            {
+                return factory.CreateLinearRing(new Coordinate[] {
+                        new Coordinate(boundingBox.xMin, boundingBox.yMax),
+                        new Coordinate(boundingBox.xMax, boundingBox.yMax),
+                        new Coordinate(boundingBox.xMax, boundingBox.yMin),
+                        new Coordinate(boundingBox.xMin, boundingBox.yMin),
+                        new Coordinate(boundingBox.xMin, boundingBox.yMax)});
+            }
+            try
+            {
+                ILinearRing shell = bboxToLinearRing(bbox);
+                ILinearRing[] tiles = bboxTiles.Select(bboxToLinearRing).ToArray();
+                var polygon = factory.CreatePolygon(shell, tiles);
+                return shell.Difference(NetTopologySuite.Operation.Union.UnaryUnionOp.Union(tiles)).IsEmpty;
+            }
+            catch(Exception e)
+            {
+                _logger.LogCritical(e, "error during linear creation");
+            }
+            return false;
```

Comment:
Thanks but I don't this works with line geometries.
Test fails with those two ones : 
`
[InlineData("SRTM_GL3", 1.5, 1.6, 44.5, 44.6, true)] // fully inside 1 tile
[InlineData("SRTM_GL3", 3.5, 4.5, 42.5, 42.6, false)] // half inside 1 tile, half with no tile
`
I'll commit a version with polygons, checking if bbox is inside the polygon = (union of DEM tiles)